core/engine/src/object/jsobject.rs

```diff
@@ -865,29 +867,92 @@ impl<T: NativeObject + ?Sized> JsObject<T> {
     }
 }
 
-impl<T: NativeObject + ?Sized> AsRef<GcRefCell<Object<T>>> for JsObject<T> {
+impl<T: NativeObject> JsObjectTyped<T> {
+    /// Creates a new `JsObjectTyped` from its root shape, prototype, and data.
+    ///
+    /// Note that the returned object will not be erased to be convertible to a
+    /// `JsValue`. To erase the pointer, call [`JsObjectTyped::upcast`].
+    pub fn new<O: Into<Option<JsObject>>>(root_shape: &RootShape, prototype: O, data: T) -> Self
+    where
+        T: Sized,
```

Comment:
`T` is always `Sized` because of the `impl` constraint so we can remove this (applies to all functions in this `impl` block) :)

```suggestion
```